### PR TITLE
USEID-377 monitor eid service

### PR DESCRIPTION
**Pull request by phuesler**
Add counters and timers to the eid service calls. I don't know if there is a better way to annotate inherited methods other than overriding and calling super.

### CODE REVIEW

File: src/main/kotlin/de/bund/digitalservice/useid/identification/IdentificationSessionController.kt [hunk 1]
```diff
@@ -36,6 +38,11 @@ class IdentificationSessionsController(
     private val eidServiceConfig: EidServiceConfiguration
 ) {
     private val log = KotlinLogging.logger {}
+    private val tcTokenCallsSuccessfulCounter: Counter =
+        Metrics.counter("eid_service.requests", "method", "get_tc_token", "status", "200")
+    private val tcTokenCallsWithErrorsCounter: Counter = Metrics.counter("eid_service.requests", "method", "get_tc_token", "status", "500")
+    private val getEidInformationCallsSuccessfulCounter: Counter = Metrics.counter("eid_service.requests", "method", "get_eid_information", "status", "200")
+    private val getEidInformationCallsWithErrorsCounter: Counter = Metrics.counter("eid_service.requests", "method", "get_eid_information", "status", "200")
```
Comment (code28) [inline src/main/kotlin/de/bund/digitalservice/useid/identification/IdentificationSessionController.kt:45]:
```suggestion
    private val getEidInformationCallsWithErrorsCounter: Counter = Metrics.counter("eid_service.requests", "method", "get_eid_information", "status", "500")
```

File: src/main/kotlin/de/bund/digitalservice/useid/eidservice/EidService.kt [hunk 2]
```diff
@@ -60,4 +63,14 @@ class EidService constructor(config: EidServiceConfiguration, private val dataGr
         request.useOperations = selector
         return request
     }
+
+    @Timed(value = "eid_service.get_tc_token")
+    public override fun getTcToken(refreshUrl: String?): TCTokenType {
+        return super.getTcToken(refreshUrl)
+    }
+
+    @Timed(value = "eid_service.get_eid_infirmation")
```
Comment (code28) [inline src/main/kotlin/de/bund/digitalservice/useid/eidservice/EidService.kt:72]:
```suggestion
    @Timed(value = "eid_service.get_eid_information")
```

**Review by mpanne — APPROVED** [review]

**Review by hannesschaletzky — APPROVED** [review]
Does it make sense to extract the "Metrics" into an own service like we did with the tracking? IdentificationSessionController.kt feels a little packed.

**Comment by phuesler** [comment]
I agree that the controller contains a lot of code and is packed. However, I think the service communication part should be extracted into its own thing to slim down the controller. The success and response counting should happen where we handle the responses, the call counting is done by the timed annotations.

**Pull request merged**